Annotato/Annotato/Persistence/PersistenceService/OfflinePersistenceService/OfflinePersistenceService+DocumentsPersistence.swift

```diff
@@ -29,7 +29,23 @@ extension OfflinePersistenceService: DocumentsPersistence {
         return await localPersistence.documents.deleteDocument(document: document)
     }
 
-    func createOrUpdateDocuments(documents: [Document]) -> [Document]? {
-        localPersistence.documents.createOrUpdateDocuments(documents: documents)
+    func createOrUpdateDocument(document: Document) async -> Document? {
+        if LocalDocumentEntityDataAccess.read(documentId: document.id,
+                                              withDeleted: true) != nil {
+            return await updateDocument(document: document)
+        } else {
+            return await createDocument(document: document)
+        }
+    }
```

Comment:
Same as the above comment.

```suggestion
    func createOrUpdateDocument(document: Document) async -> Document? {
        if await localPersistence.documents.getDocument(documentId: document.id) != nil {
            return await updateDocument(document: document)
        } else {
            return await createDocument(document: document)
        }
    }
```